src/uct/rocm/base/rocm_base.c

```diff
@@ -332,11 +368,13 @@ ucs_status_t uct_rocm_base_mem_query(uct_md_h md, const void *addr,
     }
 
     if (mem_attr_p->field_mask & UCT_MD_MEM_ATTR_FIELD_DMABUF_FD) {
-        mem_attr_p->dmabuf_fd = UCT_DMABUF_FD_INVALID;
+        uct_rocm_base_dmabuf_export(addr, length, mem_type, &dmabuf_fd,
+                                    &dmabuf_offset);
+        mem_attr_p->dmabuf_fd = dmabuf_fd;
     }
 
     if (mem_attr_p->field_mask & UCT_MD_MEM_ATTR_FIELD_DMABUF_OFFSET) {
-        mem_attr_p->dmabuf_offset = 0;
+        mem_attr_p->dmabuf_offset = dmabuf_offset;
```

Comment:
what if user asked for UCT_MD_MEM_ATTR_FIELD_DMABUF_OFFSET without UCT_MD_MEM_ATTR_FIELD_DMABUF_OFFSET?
seems offset will be always 0 in this case which is wrong..

Reply:
@yosefe So that is a good question, can this really occur? We had this discussion previously whether it is legal to ask for the offset without the handle and vice versa, see

https://github.com/openucx/ucx/pull/7847#discussion_r882048898

Reply:
Seems we didn't document or prohibit it explicitly. And I think it should be simple enough to support it.

Reply:
ok, I can add that, but basically we have to re-export the same buffer in that case, and we make the assumption that the offset obtained in the second export can be used with the dmabuf handle of the first export. I would expect it to work in most scenarios, but I am wondering whether this is truly correct in the most generic sense